pkg/policy/l4.go

```diff
@@ -230,12 +230,16 @@ func (l4 L4Filter) matchesLabels(labels labels.LabelArray) bool {
 
 // L4PolicyMap is a list of L4 filters indexable by protocol/port
 // key format: "port/proto"
-type L4PolicyMap map[string]L4Filter
+type L4PolicyMap struct {
+	Filters          map[string]L4Filter
+	SourceRuleLabels labels.Labels // The User rule labels that resulted in this Map. Can be empty
```

Comment:
I've noticed you've used `SourceRuleLabels` and `SourceRuleCount` on a previous structure (`L3PolicyMap`). Does it make sense to create a new structure and embedded that new structure in `L4PolicyMap` and `L3PolicyMap`? (This is just a thought, I'm ok as it is.)

Reply:
I think I will, since I can then attach methods to merge them and manage them. This is better since I'm violating the usage of the Labels type and I need to fix that too.